Add support for `SET AND RESET SESSION AUTHORIZATION` statements. (#10550)
crate/crate: docs/sql/statements/set-session-authorization.rst

.. highlight:: psql
.. _ref-set-session-authorization:

=======================================
``SET AND RESET SESSION AUTHORIZATION``
=======================================

> Set the session and the current user of the current session.

Reviewer: ```suggestion
Set the user of the current session.
```

.. rubric:: Table of contents

.. contents::
   :local:

Synopsis
========

::

    SET [ SESSION | LOCAL ] SESSION AUTHORIZATION username
    SET [ SESSION | LOCAL ] SESSION AUTHORIZATION DEFAULT
    RESET SESSION AUTHORIZATION

Description
===========

> These statements set or reset the session and current user of the current

Reviewer: ```suggestion
These statements set or reset the user of the
```

> session to the given or the initial session / authenticated user.

Reviewer: ```suggestion
session to the given or revert the user back to the original authenticated user.
```

> The session user can be changed only if the initial session / authenticated

Reviewer: ```suggestion
The session user can be changed only if the initial authenticated
```

> user had the superuser privileges.  Otherwise, the statement is accepted only

Reviewer: ```suggestion
user had superuser privileges.  Otherwise, the statement is only accepted
```

> if it specifies the originally authenticated user.

Reviewer: ```suggestion
if the specified username matches the originally authenticated user.
```

> Using this statement, it is possible to temporarily become an unprivileged
> user and later switch back to a superuser. In this case, the superuser must
> be the initial session / authenticated user. It can be achieved by providing
> the originally authenticated user name or the ``DEFAULT`` identifier.

Reviewer: ```suggestion
Using this statement, a superuser can temporarily become an unprivileged
user and later switch back to a superuser. The supseruser would switch using ``SET SESSION AUTHORIZATION '<impersonating_user>'`` to drop privileges and become `impersonating_user`. Later the original privileges can be restored using ``SET SESSION AUTHORIZATION <real_username>`` or ``SET SESSION AUTHORIZATION DEFAULT``.
```

``SET LOCAL`` does not have any effect on session. All ``SET LOCAL`` statements
will be ignored by CrateDB and logged with the ``INFO`` logging level.

> ``SET`` and ``SET SESSION`` may affect the current session.

Reviewer: ```suggestion
```

I think this is already clear from the previous paragraphs.
